### DISCUSSION TITLE
Sharing a hash_dir across multiple ccache versions

### DISCUSSION THREAD

**Discussion by nolange** [ccache/ccache#923] [Ideas]
Hello,

I did setup some local gitlab CI with docker runners. The idea is that all Jobs get the same ccache directory.

But that means there will be images with different versions of ccache. So my question would be:

-   Are major versions compatible, for operation concurrently on the same hash_dir?
-   Some placeholder in hash_dir would be nice that expands to an hash_dir format/protocol version. that way you
    can naively provide a directory like `/tmp/ccache-hash/%V` and have compatible ccache versions share a subdirectory.
-   Or ccache could automatically use a subdirectory `${hash_dir}/%V` (the next time an incompatible change occurs).

**Comment by jrosdahl** [comment]
Hi there,

(I assume that you mean "cache_dir" when you write "hash_dir". The [`hash_dir`](https://ccache.dev/manual/4.4.html#config_hash_dir) option is something different.)

Yes, all versions are compatible and there is no plan to break compatibility in the future either. Note: by "compatibility" I here mean that different versions can use the same cache directory without breaking each other, not that different versions always will share their results. See for instance the second bullet of the [compatibility notes for version 4.0](https://ccache.dev/releasenotes.html#_ccache_4_0).

I don't quite see the need for introducing something like `%V` in `cache_dir`.
